samcli/lib/list/resources/resource_mapping_producer.py

```diff
@@ -0,0 +1,87 @@
+"""
+The producer for the 'sam list resources' command
+"""
+from typing import Optional, Any
+import dataclasses
+import logging
+import yaml
+
+from botocore.exceptions import ClientError, NoCredentialsError
+from samtranslator.translator.managed_policy_translator import ManagedPolicyLoader
+from samtranslator.translator.arn_generator import NoRegionFound
+
+from samcli.commands.list.exceptions import SamListLocalResourcesNotFoundError, SamListUnknownClientError
+from samcli.lib.list.list_interfaces import Producer
+from samcli.lib.list.resources.resources_def import ResourcesDef
+from samcli.lib.translate.sam_template_validator import SamTemplateValidator
+from samcli.lib.providers.sam_stack_provider import SamLocalStackProvider
+from samcli.commands.translate.translate_utils import read_sam_file
+from samcli.lib.translate.exceptions import InvalidSamDocumentException
+from samcli.commands.local.cli_common.user_exceptions import InvalidSamTemplateException
+from samcli.commands.exceptions import UserException
+
+LOG = logging.getLogger(__name__)
+
+
+class ResourceMappingProducer(Producer):
+    def __init__(
+        self,
+        stack_name,
+        output,
+        region,
+        profile,
+        template_file,
+        cloudformation_client,
+        iam_client,
+        mapper,
+        consumer,
+    ):
+        self.stack_name = stack_name
+        self.output = output
+        self.region = region
+        self.profile = profile
+        self.template_file = template_file
+        self.cloudformation_client = cloudformation_client
+        self.iam_client = iam_client
+        self.mapper = mapper
+        self.consumer = consumer
+
+    def get_translated_dict(self, template_file_dict: dict) -> Optional[Any]:
```

Comment:
Let's include docstrings for all new methods please. Also for the return type, let's use `Dict[Any, Any]` as that's a little more accurate.